### PR TITLE
Add Dashboard reporter

### PR DESCRIPTION
**Pull request by leobenkel**
#### What it does

Add a dashboard reporter

#### How it works

hit the dashboard api reporter URL and post the results when test is completed

### CODE REVIEW

**Comment by leobenkel** [comment]
@hugo-vrijswijk you can take over to pass the value down to the reporter

**Comment by hugo-vrijswijk** [comment]
Awesome! Thanks a lot. I'll take a closer look at it tomorrow (EU time here)

File: core/src/main/scala/stryker4s/report/DashboardReporter.scala [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  }
+
+  private val DashboardRootURL: String = "https://dashboard.stryker-mutator.io"
+  private val DashboardURL: String = s"$DashboardRootURL/api/reports"
+
+  def buildScoreResult(input: MutantRunResults): StrykerDashboardReport = {
+    StrykerDashboardReport(
+      "PLACEHOLDER_API_KEY", // TODO: Get apikey
+      "PLACEHOLDER_REPO_SLUG", // TODO: Get repo slug
+      "PLACEHOLDER_BRANCH_NAME", // TODO: Get branch name
```
Comment (leobenkel) [inline core/src/main/scala/stryker4s/report/DashboardReporter.scala]: @hugo-vrijswijk you just have to fill those values and I think we are done.

**Comment by hugo-vrijswijk** [comment]
@leobenkel I've made some changes, could you do a review of them?

File: core/src/main/scala/stryker4s/config/ReporterType.scala [hunk 2]
```diff
@@ -18,4 +20,9 @@ case object JsonReporterType extends ReporterType {
 
 case object DashboardReporterType extends ReporterType {
   override val name: String = "dashboard"
+
+  def unapply(reporterType: ReporterType)(implicit config: Config): Option[DashboardReporter] = reporterType match {
```
Comment (leobenkel) [inline core/src/main/scala/stryker4s/config/ReporterType.scala]: missing returned type. It speed up compilation

File: core/src/main/scala/stryker4s/report/DashboardReporter.scala [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+  }
+
+  def writeReportToDashboard(url: String, report: StrykerDashboardReport): WebResponse = {
+    webIO.postRequest(url, report.toJson)
+  }
+
+  override def reportRunFinished(runResults: MutantRunResults): Unit = {
+    val response = writeReportToDashboard(dashboardURL, buildScoreResult(runResults))
+
+    if (response.httpCode == 201) {
```
Comment (hugo-vrijswijk) [inline core/src/main/scala/stryker4s/report/DashboardReporter.scala]: Should be something like "is in OK range (2xx)"

File: core/src/main/scala/stryker4s/http/webIO.scala [hunk 4]
```diff
@@ -0,0 +1,24 @@
+package stryker4s.http
+
+trait WebIO {
+  def postRequest(url: String, content: String): WebResponse
+}
+
+case class WebResponse(httpCode: Int, responseBody: String)
+
+object RealHttp extends WebIO {
```
Comment (hugo-vrijswijk) [inline core/src/main/scala/stryker4s/http/webIO.scala]: Can we rename this to HttpClient it feels more natural of a name to me.

File: core/src/main/scala/stryker4s/report/DashboardReporter.scala (hunk 3 above)
Comment (hugo-vrijswijk) [reply core/src/main/scala/stryker4s/report/DashboardReporter.scala]: Fixed, now uses scalaj-http response which has `isSuccess`

**Comment by legopiraat** [comment]
Thank you @leobenkel!

**Comment by nicojs** [comment]
Awesome new feature, thanks! 👍

**Comment by leobenkel** [comment]
What version will it be part of ?

**Comment by hugo-vrijswijk** [comment]
You should already be able to use the snapshot. I'll make a release (0.5.0) soon (no promises)

**Comment by leobenkel** [comment]
> You should already be able to use the snapshot. I'll make a release (0.5.0) either soon (no promises) or tomorrow

cool thanks. I'll wait for `0.5.0` .

**Comment by hugo-vrijswijk** [comment]
@leobenkel I just [started a new release process](https://travis-ci.org/stryker-mutator/stryker4s/builds/544417659). Once that finishes and syncs to Maven Central it should be available. I'll write release notes tomorrow

**Comment by leobenkel** [comment]
@hugo-vrijswijk just wanted to let you know that the dashboard worked ! https://travis-ci.org/leobenkel/safety_plugin/builds/547924432 It was submitted successfully!
The badge got updated: https://badge.stryker-mutator.io/github.com/leobenkel/safety_plugin/master 

It was pretty straightforward to use ! The only issue I had was the size of the log exploded in Travis face so I had to change the log level to Error only to be able to fully run the entire thing.

**Comment by hugo-vrijswijk** [comment]
Awesome to hear! I've noticed that on CI tests of mutation runs will also be logged, even though they should only be logged on debug. So I'll have to look into that. Thanks for the feedback!

**Pull request merged**